### ISSUE TITLE
Uncaught TypeError: Cannot add property segmentsIntersect, object is not extensible

### ISSUE THREAD

**Issue by dumrakoti** [StevenHeinrich/ember-leaflet-draw#5]
While installing ember-leaflet-draw  its show error `Uncaught TypeError: Cannot add property segmentsIntersect, object is not extensible`

leaflet version: `^3.0.12`
Ember version: `2.13.3`

**Comment by StevenHeinrich** [comment]
Based on what you listed, it sounds like you are on:

> Ember-Leaflet: ^3.0.12
> Ember: 2.13.3

Additional questions for you, related to version information:

1. What version of Ember-Leaflet-Draw? (my assumption is older than `0.6.1`)
2. What version of LeafletJS do you have installed? (my assumption is `1.1.0`, not `1.0.3`)

---

My first guess at the problem (prior to having your additional feedback):

It may have to do with you recently doing an `npm install` while using a version of Ember-Leaflet-Draw that is older than `0.6.1`.

With `Ember-Leaflet-Draw@0.6.1`, the npm dependency for LeafletJs changed, locking it to `Leaflet@1.0.3`. Leaflet just upgraded to 1.1.0 on June 27, 2017 (http://leafletjs.com/2017/06/27/leaflet-1.1.0.html). I seemed to be having errors, like the one you mentioned.

A few things with those errors prompted me to simply restrict the Leaflet dependency version to 1.0.3, **for now** (short term solution that works as long as you don't need anything new from Leaflet@1.1.0). I mentioned a little bit about this version lock in a previous, but unrelated,  issue (https://github.com/StevenHeinrich/ember-leaflet-draw/issues/4#issuecomment-311658776).

1. The primary Ember Addon (Ember-Leaflet), has not yet had a build on TravisCI since prior to the latest release of LeafletJS 1.1.0. This means that it has not shown a successful build that uses `Leaflet@1.1.0`.
2. I have not had time to look into the details.
3. Optimally, I will piggyback off Ember-Leaflet... after they run into this issue and determine their course or action... as I am fairly confident that they will have similar trouble. At that point, Ember-Leaflet-Draw can follow the course of action taken by its parent addon.

---

### tl;dr
My hope is that your current needs don't rely on any features new to the latest Leaflet@1.0.1 release. If that is true, I am optimistic that updating Ember-Leaflet-Draw to at least 0.6.1 will help to clear that error. If you see trouble after upgrading, double check that your installed version of LeafletJS is `1.0.3`.

Please, let me know if this helps or not... or if you come across any other errors in the process.

**Comment by Krishna01992** [comment]
ember-leaflet-draw version :  `0.7.0`

I have checked in `1.1.0` and `1.0.3` version of leafletJS and still shows same errors. And also checked in `0.6.1` version of ember-leaflet-draw, still same error.

`Uncaught TypeError: Cannot add property segmentsIntersect, object is not extensible
    at Object.extend`

**Comment by dumrakoti** [comment]
Thanks @StevenHeinrich 
Its working now

**Comment by rikee312** [comment]
@Krishna01992  I solved this issue with
ember-cli: `2.13.3`
ember-leaflet: `3.0.12`
leaflet: `1.0.3`
ember-leaflet-draw:` 0.6.1`
